crates/sui-rosetta/src/types.rs

```diff
@@ -556,10 +556,10 @@ pub struct ConstructionPreprocessRequest {
     pub metadata: Option<PreprocessMetadata>,
 }
 
-#[derive(Serialize, Deserialize)]
-pub enum PreprocessMetadata {
-    PaySui,
-    Delegation,
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PreprocessMetadata {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub budget: Option<u64>,
```

Comment:
I believe this is not being use anywhere, maybe can remove?

Reply:
You mean the just the derive Debug part? If so, yes I can remove it.